build/rewrite.go

```diff
@@ -54,12 +54,59 @@ func allowedSort(name string) bool {
 	return false
 }
 
-// Rewrite applies the high-level Buildifier rewrites to f, modifying it in place.
+// RewristeSet will contain rewrites to be applied.
+type RewriteSet struct {
+	Names []string
+}
+
+// Rewrites controls the rewrites to be applied.
+//
+// If non-nil, the rewrites with the specified names will be run. If
+// nil, a default set of rewrites will be used that is determined by
+// the type (BUILD vs default starlark) of the file being rewritten.
+type Rewriter struct {
+	Rewrites                        *RewriteSet
+	LabelArg                        map[string]bool
+	Denylist                        map[string]bool
+	SortableListArg                 map[string]bool
+	SortDenyList                    map[string]bool
+	SortAllowList                   map[string]bool
+	NamePriority                    map[string]int
+	StripLabelLeadingSlashes        bool
+	ShortenAbsoluteLabelsToRelative bool
+}
+
 func Rewrite(f *File) {
-	for _, r := range rewrites {
-		if !disabled(r.name) {
-			if f.Type&r.scope != 0 {
-				r.fn(f)
+	var rewriter = &Rewriter{
+		LabelArg:                        tables.IsLabelArg,
+		Denylist:                        tables.LabelDenylist,
+		SortableListArg:                 tables.IsSortableListArg,
+		SortDenyList:                    tables.SortableDenylist,
+		SortAllowList:                   tables.SortableAllowlist,
+		NamePriority:                    tables.NamePriority,
+		StripLabelLeadingSlashes:        tables.StripLabelLeadingSlashes,
+		ShortenAbsoluteLabelsToRelative: tables.ShortenAbsoluteLabelsToRelative,
+	}
+	rewriter.Rewrite(f)
+}
+
+func (w *Rewriter) Rewrite(f *File) {
+	if w.Rewrites == nil {
+		for _, r := range rewrites {
+			if !disabled(r.name) {
+				if f.Type&r.scope != 0 {
+					r.fn(w, f)
+				}
+			}
+		}
+	} else {
+		for _, r := range rewrites {
+			if !disabled(r.name) {
+				for _, n := range w.Rewrites.Names {
```

Comment:
Do you need the outer for-loop here? Why not just iterate `w.Rewrites.Names`?

Reply:
Also you have two chunks of code that are almost duplicates, maybe it makes sense to create a local variable that's either equal to `w.Rewrites.Names` or `rewrites` and iterate it just once?

Reply:
Made code smaller and more legible